gitlab/base.py

```diff
@@ -217,6 +217,15 @@ def get_id(self) -> Any:
             return None
         return getattr(self, self._id_attr)
 
+    @property
+    def encoded_id(self) -> Any:
+        """Ensure that the ID is url-encoded so that it can be safely used in a URL
+        path"""
+        obj_id = self.get_id()
+        if isinstance(obj_id, str):
+            obj_id = gitlab.utils.EncodedId(obj_id)
```

Comment:
question: do we need this check here if `EncodedId` will handle it anyway? I guess there's maybe a performance penalty when we receive `int` but not sure it's noticeable.

Same for others below, and seems like you did remove the check in some of them, right?

Reply:
I see. Could we maybe just return int in `__new__` if we receive int, essentially making it a no-op? It would also simplify `__init__` then, as it would never get called for ints. Like discussed here.

https://stackoverflow.com/a/2491881

I feel like this way we'd also stick the closest to the current behavior, possibly avoiding some issues.